src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/VbClassLibraryMyDotAnalyzer.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.UpgradeAssistant.Dependencies;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Packages.Analyzers
+{
+    /// <summary>
+    /// This analyzer will modify the vbproj of class libraries to resolve compilation errors:
+    /// 1. Adding System.Configuration.ConfigurationManager resolves the error from Settings.Designer.vb
+    /// 2. Adding <VBRuntime>Embed</VBRuntime> resolves the error from the vb compiler
+    ///     Per https://github.com/dotnet/runtime/issues/30478#issuecomment-521270193.
+    /// </summary>
+    public class VbClassLibraryMyDotAnalyzer : IDependencyAnalyzer
+    {
+        private const string SystemConfigurationPackageName = "System.Configuration.ConfigurationManager";
+
+        private readonly IPackageLoader _packageLoader;
+        private readonly ILogger<VbClassLibraryMyDotAnalyzer> _logger;
+
+        public string Name => nameof(VbClassLibraryMyDotAnalyzer) + " reference analyzer";
+
+        public VbClassLibraryMyDotAnalyzer(IPackageLoader packageLoader, ILogger<VbClassLibraryMyDotAnalyzer> logger, ITargetFrameworkMonikerComparer tfmComparer)
+        {
+            _packageLoader = packageLoader ?? throw new ArgumentNullException(nameof(packageLoader));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async Task AnalyzeAsync(IProject project, IDependencyAnalysisState state, CancellationToken token)
+        {
+            if (project is null)
+            {
+                throw new ArgumentNullException(nameof(project));
+            }
+
+            if (state is null)
+            {
+                throw new ArgumentNullException(nameof(state));
+            }
+
+            if (!project.TargetFrameworks.Any(tfm => tfm.IsNetCore))
+            {
+                _logger.LogDebug("None of the tfms match packages from {PackageName}", SystemConfigurationPackageName);
+                return;
+            }
+
+            if (!IsVbClassLibrary(project))
+            {
+                _logger.LogDebug("{Project} is not a VB class library", project.FileInfo);
+                return;
+            }
+
+            if (string.IsNullOrWhiteSpace(project.GetFile().GetPropertyValue("VBRuntime")))
```

Comment:
nit: use a local variable for the file (currently it's backed by the same object, but it would be great to get a snapshot at some point)